sktime/classification/distance_based/tests/test_shape_dtw.py

```diff
@@ -0,0 +1,172 @@
+import numpy as np
+import pytest
+
+from sktime.classification.distance_based._shape_dtw import ShapeDTW
+from sktime.utils._testing import generate_df_from_array
+
+from sktime.transformers.series_as_features.paa_multivariate \
+    import PAA_Multivariate
+from sktime.transformers.series_as_features.dwt import DWT
+from sktime.transformers.series_as_features.slope import Slope
+from sktime.transformers.series_as_features.derivative import Derivative
+from sktime.transformers.series_as_features.hog1d import HOG1D
+
+from sktime.datasets import load_gunpoint
+
+
+# Check that exception is raised for bad subsequence length.
+@pytest.mark.parametrize("bad_subsequence_length", ['str', 0.5, -1, -0.5, {}])
```

Comment:
it would help to leave comments next to the bad input types (e.g. string, float, negative int, negative float, empty dict) and to mention the expected input type